listenbrainz/db/user_timeline_event.py

```diff
@@ -168,3 +190,66 @@ def get_user_notification_events(user_id: int, count: int = 50) -> List[UserTime
         event_type=UserTimelineEventType.NOTIFICATION,
         count=count
     )
+
+def hide_user_timeline_event(user_id: int, event_type: UserTimelineEventType, event_id: int) -> bool:
+    """ Adds events that are to be hidden """
+    try:
+        with db.engine.connect() as connection:
+            result = connection.execute(sqlalchemy.text('''
+                INSERT INTO hide_user_timeline_event (user_id, event_type, event_id)
+                    SELECT :user_id, :event_type, :event_id WHERE NOT EXISTS
```

Comment:
Why not a simple insert? If you are trying to avoid duplicates, create a UNIQUE index and use ON CONFLICT DO NOTHING. We have some existing examples in the codebase for this, you can find them by searching for this clause.